timesync/manual_clock.go

```diff
@@ -0,0 +1,28 @@
+package timesync
+
+import "time"
+
+type ManualClock struct {
+	*Ticker
+	genesisTime time.Time
+}
+
+func (t *ManualClock) Tick() {
+	t.Notify()
+}
+
+func NewManualClock(genesisTime time.Time) *ManualClock {
```

Comment:
Genesis time shouldn't be a parameter into this func since this does not depend on time

Reply:
It probably previously had it from the ticker.
Now they are separated.
Leaving it as is.